dali/kernels/audio/mel_scale/mel_filter_bank_gpu.cu

```diff
@@ -0,0 +1,311 @@
+  int64_t fftbin_end = interval_ends[mel_bin];
+  const T *in =  in_frame + fftbin * fft_stride + fft_shift;
+  for (; fftbin <= fftbin_end; ++fftbin, in += fft_stride) {
+    auto weight_up = T(1) - weights_down[fftbin];
+    if (normalize) weight_up *= norm_factors[mel_bin];
+    out += *in * weight_up;
+  }
+  // filter down
+  fftbin = interval_ends[mel_bin] + 1;
+  fftbin_end = (mel_bin < mel_bins - 1) ? interval_ends[mel_bin + 1] : fftbin_last;
+  in =  in_frame + fftbin * fft_stride + fft_shift;
+  for (; fftbin <= fftbin_end; ++fftbin, in += fft_stride) {
+    auto weight_down = weights_down[fftbin];
+    if (normalize) weight_down *= norm_factors[mel_bin + 1];
+    out += *in * weight_down;
+  }
+  return out;
+}
+
+// For layouts where the frequency is not the innermost dimension, data is flattened into
+// 3 dimensions - frame, frequency, time
+// Every frame is treated as independent two-dimensional sample
+template <typename T>
+__global__ void MelFilterBankKernel(const BlockDesc<T> *block_desc,
+                                    const T *weights_down, const int64_t *interval_ends,
+                                    bool normalize, const T *norm_factors,
+                                    int64_t fftbin_first, int64_t fftbin_last,
+                                    int64_t mel_bins) {
+  auto block_id = blockIdx.x;
+  const T *in_frame = block_desc[block_id].in_frame;
+  T *out_frame = block_desc[block_id].out_frame;
+  int64_t mel_bin = blockIdx.y * kBlockDim2 + threadIdx.y;
+  if (mel_bin >= mel_bins) return;
+  int64_t window = block_desc[block_id].start_window + threadIdx.x;
+  int64_t nwindows = block_desc[block_id].frame_nwindows;
+  if (window >= nwindows) return;
+  T *out = out_frame + mel_bin * nwindows + window;
+  *out = calcMel(in_frame, mel_bin,
+                 weights_down, interval_ends, normalize, norm_factors,
+                 fftbin_first, fftbin_last,
+                 nwindows, window, mel_bins);
+}
+
+// For layouts with the innermost frequency dimension, data is flattened
+// to two dimensions - time, frequency
+template <typename T>
+__global__ void MelFilterBankKernelInnerFft(const BlockDesc<T> *block_desc,
+                                            const T *weights_down, const int64_t *interval_ends,
+                                            bool normalize, const T *norm_factors,
+                                            int64_t fftbin_first, int64_t fftbin_last,
+                                            int64_t mel_bins, int64_t fftdim) {
+  auto block_id = blockIdx.x;
+  auto idx = block_desc[block_id].block_start + threadIdx.x;
+  if (idx >= block_desc[block_id].out_frame_size) return;
+  auto window = idx / mel_bins;
+  auto mel_bin = idx % mel_bins;
+  const T *in = block_desc[block_id].in_frame;
+  T *out =  block_desc[block_id].out_frame;
+  *(out + idx) = calcMel(in + window * fftdim, mel_bin, weights_down, interval_ends,
```

Comment:
Nitpick - I would align the argument formatting to L89, it would be easier to compare this two.